modules/c++/re/tests/benchmark_regex.cpp

```diff
@@ -0,0 +1,96 @@
+
+#include <string>
+#include <fstream>
+
+#include <import/re.h>
+#include <sys/StopWatch.h>
+#include <sys/LocalDateTime.h>
+
+// First just benchmark the Regex creation time (including compile)
+double BM_RegexCreation(uint64_t numIterations)
+{
+    sys::RealTimeStopWatch sw;
+
+    sw.start();
+    for (uint64_t ii = 0; ii < numIterations; ++ii)
+    {
+        re::Regex regex("beam(Id|String)");
+    }
+    double elapsedTimeMS = sw.stop();
+
+    return elapsedTimeMS / numIterations;
+}
+
+
+// Now benchmark the actual string-matching
+double BM_RegexMatch(uint64_t numIterations, const std::string& fileString)
+{
+    sys::RealTimeStopWatch sw;
+
+    // should match multiple times
+    re::Regex regex("beam(Id|String)");
+
+    // should never match
+    //re::Regex regex("foobarfighters");
+
+    sw.start();
+    for (uint64_t ii = 0; ii < numIterations; ++ii)
+    {
+        regex.matches( fileString );
+    }
+    double elapsedTimeMS = sw.stop();
+
+    return elapsedTimeMS / numIterations;
+}
+
+
+int main(int argc, char** argv)
+{
+    if (argc < 3)
+    {
+        std::cerr << "Usage: ./bm_regex inputFile numIterations" << std::endl;
+        return 1;
+    }
+
+    uint64_t numIterations = std::stoi(argv[2]);
+
+    sys::LocalDateTime ldt;
+    std::cout << ldt.format(std::string("%Y-%m-%d %H:%M:%S")) << std::endl;
+    
+    // Open our text file and feed it into the static buffer
+    std::ifstream bigFin(argv[1]);
+    if (!bigFin.is_open())
+    {
+        std::cerr << "Error opening text file!" << std::endl;
+        return 2;
+    }
+    
+    size_t size = bigFin.tellg();
+    std::string fileString(size, '\0');
+    
+    bigFin.seekg(0);
+    bigFin.read(&fileString[0], size);
```

Comment:
It's not guaranteed that this is safe to write into is it (even though presumably the underlying implementation is a char array)?  I've always used a `std::vector<char>` for this sort of thing...